analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp6.cs

```diff
@@ -52,11 +53,11 @@ public void TryCatch4(object o)
             o = null;
             try
             {
-                var a = o?.ToString();
+                var a = o?.ToString();  // Compliant
             }
             catch (Exception e) when (e.Message != null)
             {
-                var b = o.ToString(); // FIXME Non-compliant, o could be null here
+                var b = o.ToString(); // Compliant FIXME (was Non-compliant before)
```

Comment:
My suspicion is, that the engine is smart enough to see that the catch block is unreachable (There is no code path that can throw). But maybe I'm wrong.

Reply:
An `Console.WriteLine("");` did the trick.